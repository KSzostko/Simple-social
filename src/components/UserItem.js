import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledListItem = styled.li`
    display: grid;
    grid-template-rows: auto auto 1fr auto;
    padding: 20px;
    border: 1px solid #000;
`;

const StyledInfoList = styled.ul`
    display: flex;
    flex-direction: column;
    list-style: none;
    padding: 0;
    margin-bottom: 20px;
    font-size: ${props => props.contact ? '12px' : '16px'};
    font-weight: ${props => props.contact ? 300 : 400};
`;

const StyledLink = styled(Link)`
    background-color: #e4bb7a;
    padding: 15px 40px;
    font-size: 1.25rem;
    color: #fff;
    border-style: none;
    cursor: pointer;
    transition: 0.25s;
    text-decoration: none;
    text-align: center;

    &:hover {
        background-color: rgba(187, 144, 75, 0.9);
    }
`;

const UserItem = ({ data }) => {
    const { id, name, email, phone, website, company } = data;
    
    return (
        <StyledListItem>
            <h2>{name}</h2>
            <StyledInfoList contact>
                <li>{email}</li>
                <li>{phone}</li>
                <li>{website}</li>
            </StyledInfoList>
            <StyledInfoList>
                <li>{company.name}</li>
                <li>{company.catchPhrase}</li>
                <li>{company.bs}</li>
            </StyledInfoList>
            <StyledLink
                to={{
                    pathname: `/user/${id}`,
                    state: {
                        name: name,
                    }
                }}
            >
                Details
            </StyledLink>
        </StyledListItem>
    );
};

export default UserItem;