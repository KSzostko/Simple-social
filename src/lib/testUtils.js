const fakePost = () => ({
    userId: 1,
    postId: 2,
    title: 'Post title',
    body: 'Post body',
});

const fakeComment = () => ({
    postId: 1,
    id: 1,
    name: 'Comment name',
    email: 'user@gmail.com',
    body: 'Comment body',
});

const fakeUser = () => ({
    id: 1,
    name: 'John Doe',
    email: 'jdoe@gmail.com',
    phone: '123-456-789',
    website: 'hildegard.org',
    company: {
        name: 'Company Name',
        catchPhrase: 'phrase',
        bs: 'bs text',
    }
});

export { fakePost, fakeComment, fakeUser };