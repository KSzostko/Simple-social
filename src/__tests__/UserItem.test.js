import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { fakeUser } from '../lib/testUtils';
import UserItem from '../components/UserItem';

const fakeData = {
    data: { ...fakeUser() }
};

describe('<UserItem />', () => {
    it('renders and matches the snapshot', () => {
        const { container } = render(<UserItem {...fakeData} />, {
            wrapper: MemoryRouter,
        });

        expect(container.firstChild).toMatchSnapshot();
    });
});