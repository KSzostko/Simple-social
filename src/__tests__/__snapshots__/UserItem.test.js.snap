// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> renders and matches the snapshot 1`] = `
<li
  class="sc-AxjAm gSFqbq"
>
  <h2>
    John Doe
  </h2>
  <ul
    class="sc-AxirZ dzmujv"
  >
    <li>
      jdoe@gmail.com
    </li>
    <li>
      123-456-789
    </li>
    <li>
      hildegard.org
    </li>
  </ul>
  <ul
    class="sc-AxirZ kEnYeA"
  >
    <li>
      Company Name
    </li>
    <li>
      phrase
    </li>
    <li>
      bs text
    </li>
  </ul>
  <a
    class="sc-AxiKw jKyKl"
    href="/user/1"
  >
    Details
  </a>
</li>
`;
